fdbserver/ptxn/TLogGroupVersionTracker.actor.cpp

```diff
@@ -60,9 +65,10 @@ std::map<TLogGroupID, Version> TLogGroupVersionTracker::updateGroups(const std::
 		it->second = commitVersion;
 	}
 
-	if (SERVER_KNOBS->INSERT_EMPTY_TRANSACTION || SERVER_KNOBS->BROADCAST_TLOG_GROUPS) {
+	if (SERVER_KNOBS->INSERT_EMPTY_TRANSACTION || SERVER_KNOBS->BROADCAST_TLOG_GROUPS ||
+	    returnAllGroups == UpdateAllGroups::True) {
 		for (auto& [gid, version] : versions) {
```

Comment:
Is there a place where `versions` is initialized for `all` tlog groups for every resolver?

Reply:
Master sends all TLog groups to all resolvers during resolvers' initialization https://github.com/jzhou77/foundationdb/blob/815fc051779736eefdb075ad4e27d869028990dc/fdbserver/masterserver.actor.cpp#L1050-L1053. Resolvers add new groups here: https://github.com/jzhou77/foundationdb/blob/815fc051779736eefdb075ad4e27d869028990dc/fdbserver/Resolver.actor.cpp#L137

Reply:
It seems that it is the side effect of the initial txn resolving containing all the groups, so that the versionTracker has all the groups. By just reading the resolver's code, it's unclear how does the versionTracker know all the groups. I think some comments either in master, or in the resolver are needed to explain how versionTracker is initialized with all the groups.

Reply:
Good idea. Add comments to `Resolver::versionTracker`.